### PR TITLE
Sample

### PR DESCRIPTION
**Pull request by mphammer**
A Sample template and bash script to perform our current process of adding new Controllers/CRDs to the Synopsys-Operator.

### CODE REVIEW

File: install/common/default-values.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "Namespace": "synopsys-operator",
-  "Image": "docker.io/blackducksoftware/synopsys-operator:2019.2.0-RC",
+  "Image": "gcr.io/saas-hub-stg/blackducksoftware/synopsys-operator:sample",
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply install/common/default-values.json]: yup.. It should be master

File: install/kube/cleanup.sh [hunk 2]
```diff
@@ -7,6 +7,7 @@ kubectl delete ns $NS
 kubectl delete crd alerts.synopsys.com
 kubectl delete crd blackducks.synopsys.com
 kubectl delete crd opssights.synopsys.com
+kubectl delete crd samples.synopsys.com
```
Comment (msenmurugan) [inline install/kube/cleanup.sh]: remove the sample crd from cleanup script

File: install/openshift/cleanup.sh [hunk 3]
```diff
@@ -7,6 +7,7 @@ oc delete ns $NS
 oc delete crd alerts.synopsys.com
 oc delete crd blackducks.synopsys.com
 oc delete crd opssights.synopsys.com
+oc delete crd samples.synopsys.com
```
Comment (msenmurugan) [inline install/openshift/cleanup.sh]: remove the sample crd from cleanup script

File: pkg/sample/sampledeployment.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	pod.AddContainer(specConfig.getSampleContainer())
+
+	return pod
+}
+
+// Returns a Horizon Container for the Sample
+func (specConfig *SampleSpecConfig) getSampleContainer() *components.Container {
+	container := components.NewContainer(horizonapi.ContainerConfig{
+		Name:  specConfig.config.Namespace,
+		Image: "registry.hub.docker.com/vasiliys/public-test:latest",
```
Comment (jdartigalongue) [inline pkg/sample/sampledeployment.go]: What's does this container do? Should we maybe use something more generic?

File: cmd/operator/run.go [hunk 5]
```diff
@@ -55,15 +56,20 @@ func kill(stopCh chan struct{}) {
...
 func runProtoform(configPath string) {
+	// Add controllers to the Protoform Deployer
 	deployer, err := protoform.NewController(configPath)
 	if err != nil {
 		panic(err)
 	}
 
 	stopCh := make(chan struct{})
 
+	sampleController := sample.NewCRDInstaller(deployer.Config, deployer.KubeConfig, deployer.KubeClientSet, bdutil.GetSampleDefaultValue(), stopCh)
```
Comment (jdartigalongue) [inline cmd/operator/run.go]: This should be commented out. I don't think that we want to deploy the sample CRD with every installation of the synopsys operator.

File: pkg/sample/sampledeployment.go (hunk 4 above)
Comment (mphammer) [reply pkg/sample/sampledeployment.go]: This is an image Vasiliy made that repeatedly prints a message. I can change it to something more generic if you have something in mind. 
@VSharapov

Comment (rrati) [reply pkg/sample/sampledeployment.go]: If that image isn't specifically needed then I'd use one from the common images in the community.  Like a redis or something else small.

File: install/common/default-values.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (mphammer) [reply install/common/default-values.json]: I changed it to master

File: .gitignore [hunk 6]
```diff
@@ -1,6 +1,7 @@
 cmd/operator-build
 install/secured-registry/dockercfg.json
 cmd/blackduckctl/blackduckctl
```
[the remark this answers is not in the document]
Comment (mphammer) [reply .gitignore]: I removed it form the gitignore

File: install/kube/cleanup.sh (hunk 2 above)
Comment (mphammer) [reply install/kube/cleanup.sh]: I removed it

File: install/openshift/cleanup.sh (hunk 3 above)
Comment (mphammer) [reply install/openshift/cleanup.sh]: I removed it

File: pkg/sample/sampledeployment.go (hunk 4 above)
Comment (mphammer) [reply pkg/sample/sampledeployment.go]: @msenmurugan suggested I change it to <<image_name>> and tell the user to change it if they want to in the docs.

File: cmd/operator/run.go (hunk 5 above)
Comment (mphammer) [reply cmd/operator/run.go]: I commented this out. I also commented out the GetSampleDefaultValue() function in apps/util/crd_defaults.go

File: examples/sample.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+	"kind": "Sample",
+	"metadata": {
+	  "clusterName": "",
+	  "name": "sample"
+	},
+	"spec": {
+	  "namespace": "sample",
+	  "samplevalue": "YAY"
+	}
+  }
```
Comment (msenmurugan) [inline examples/sample.json]: Do we need this file??

File: hack/generate-crd.sh [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+### THIS CODE IS MODIFIED FROM update-crds.sh ####################
+echo "Cloning Kubernetes Libraries..."
+mkdir -p $GOPATH/src/k8s.io
+pushd $GOPATH/src/k8s.io > /dev/null 2>&1
+  for REPO in code-generator apimachinery api
+  do
+    echo " > git clone git@github.com:kubernetes/${REPO}.git"
+    git clone git@github.com:kubernetes/${REPO}.git > /dev/null 2>&1
+    pushd $REPO > /dev/null 2>&1
+      git pull > /dev/null 2>&1
```
Comment (msenmurugan) [inline hack/generate-crd.sh]: Can you please make the similar change that I did in update-crd.sh file?

File: examples/sample.json (hunk 7 above)
Comment (mphammer) [reply examples/sample.json]: I removed the file

File: hack/generate-crd.sh (hunk 8 above)
Comment (mphammer) [reply hack/generate-crd.sh]: I updated the code

**Pull request merged**